### PR TITLE
Add [String: RawRepresentable] support

### PR DESCRIPTION
**Pull request by yonaskolb**
This rounds out the supported types

### CODE REVIEW

**Review by lucianomarisi — APPROVED** [review]

**Comment by lucianomarisi** [comment]
Nice! Could you update the README and CHANGELOG for completeness?

File: Tests/JSONUtilitiesTests/InlineDecodingTests.swift [hunk 1]
```diff
@@ -78,6 +78,24 @@ class InlineDecodingTests: XCTestCase {
     }
   }
 
+  func test_decodingMandatoryEnumDictionary_withKey() {
+    let dictionary: JSONDictionary = ["enums": ["value1": "one", "value2": "!@1", "value3": "two"]]
+
+    let decodedEnums: [String: MockParent.MockEnum] = try! dictionary.json(atKeyPath: "enums")
```
[the remark this answers is not in the document]
Comment (yonaskolb) [reply Tests/JSONUtilitiesTests/InlineDecodingTests.swift:84]: Was just following the style of the other tests

**Comment by yonaskolb** [comment]
Sorry, committed the readme and changelog in the wrong pull request. Fixed now!

File: Tests/JSONUtilitiesTests/InlineDecodingTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucianomarisi) [reply Tests/JSONUtilitiesTests/InlineDecodingTests.swift:84]: Makes sense, at some point we should clean this things, something for the future...

**Review by lucianomarisi — APPROVED** [review]

**Comment by yonaskolb** [comment]
Once this is merged is there anything else you would want for the 4.0 release?

Keypath escaping and InvalidItemBehaviour monitoring are purely additive and non breaking changes, so they could come later

**Comment by lucianomarisi** [comment]
Sounds good, I'll release V4 this weekend. Sorry for the delays, I've been busy/distracted.

**Pull request merged**